x/auth/module.go

```diff
@@ -182,6 +182,24 @@ func (am AppModule) TxValidator(ctx context.Context, tx transaction.Tx) error {
 // ConsensusVersion implements appmodule.HasConsensusVersion
 func (AppModule) ConsensusVersion() uint64 { return ConsensusVersion }
 
+// BeginBlock implements appmodule.AppModule
+func (am AppModule) BeginBlock(ctx context.Context) error {
+	v, err := am.accountKeeper.Params.Get(ctx)
+	if err != nil {
+		return err
+	}
+	// Set block Params every being block to be used in antehandler
+	am.accountKeeper.Environment.DecodedStore.OpenCache(ctx).Set(types.ParamsKey, v)
```

Comment:
Does it work with branching? If it's called in a branched code, and the state is reverted, does this cache revert too?

Reply:
it would not, i can look into this.